### PR TITLE
complete out the os.sched_getaffinity error set

### PR DESCRIPTION
**Pull request by komuw**
-  complete out the os.sched_getaffinity error set
- add doc string for sched_getaffinity
- add test for os.sched_getaffinity

### CODE REVIEW

File: lib/std/os/test.zig [hunk 1]
```diff
@@ -601,3 +601,7 @@ test "getrlimit and setrlimit" {
     const cpuLimit = try os.getrlimit(.CPU);
     try os.setrlimit(.CPU, cpuLimit);
 }
+
+test "os.sched_getaffinity" {
+    testing.expectError(error.NoThread, os.sched_getaffinity(std.math.maxInt(i32)));
```
Comment (komuw) [inline lib/std/os/test.zig:606]: My assumption here is that we will never have as many threads as `maxInt(i32)` when running tests.    
But there is a good probability that I'm wrong/

File: lib/std/thread.zig [hunk 2]
```diff
@@ -63,10 +63,10 @@ pub const Thread = struct {
...
         } else
             return switch (std.Target.current.os.tag) {
-            .linux => os.linux.gettid(),
-            .windows => windows.kernel32.GetCurrentThreadId(),
-            else => @compileError("Unsupported OS"),
-        };
+                .linux => os.linux.gettid(),
+                .windows => windows.kernel32.GetCurrentThreadId(),
+                else => @compileError("Unsupported OS"),
+            };
```
Comment (komuw) [inline lib/std/thread.zig:69]: I ran `zig fmt` on this file and this change happened.

File: lib/std/os/test.zig (hunk 1 above)
Comment (komuw) [reply lib/std/os/test.zig:606]: On windows and Osx, this is failing with `error: use of undeclared identifier 'cpu_set_t'` . 

Should I restrict this test to linux only?

Comment (LemonBoy) [reply lib/std/os/test.zig:606]: > On windows and Osx, this is failing with

Welcome to the world of cross-platform non-posix abstractions (and yet another opportunity to plug #6600):
* NetBSD has `pthread_setaffinity_np`/`pthread_getaffinity_np`
* FreeBSD has `cpuset_getaffinity`/`cpuset_setaffinity`
* macOS has no direct equivalent, `thread_policy_get`/`thread_policy_set` let you get/set the affinity tag for a given thread but that's it
* OpenBSD has no such API (cc @semarie)

> My assumption here is that we will never have as many threads as maxInt(i32) when running tests.

That's a pretty safe assumption (why are PIDs signed???), spawning that many threads requires an ungodly amount of memory heh

File: lib/std/thread.zig (hunk 2 above)
Comment (LemonBoy) [reply lib/std/thread.zig:69]: No worries, it happens from time to time when people forget to run `zig fmt`

**Comment by semarie** [comment]
> * OpenBSD has no such API (cc @semarie)

hep. I confirm that it doesn't exist on OpenBSD.

Depending your intent for `os.sched_getaffinity()` interface, such cases could be:
- a `@compileError`, but it means any use of it will break compilation on such platforms
- a nop (returning a constant on `os.sched_getaffinity`, and
  silencious ignore value setting on `os.sched_setaffinity`).

the linux man page documents it as "setting the CPU affinity mask can
be used to obtain performance benefits". so I prefer a functional
program but with less performance instead of an unbuildable program.

File: lib/std/os/test.zig (hunk 1 above)
Comment (komuw) [reply lib/std/os/test.zig:611]: I have skipped the test for non-linux OSes. I have also opened https://github.com/ziglang/zig/issues/6907 to implement `sched_getaffinity` for other OSes. I'll attempt to do that in a follow up PR, if no one else will have done so already.

**Comment by andrewrk** [comment]
The code is correct in master branch; those error codes are not reachable, except by programmer error.

**Pull request closed without merging**